Author: Andreas Tille <tille@debian.org>
Last-Update: Wed, 25 May 2016 15:38:07 +0200
Description: Adding two files that are part of clapack source
 The clapack archive that can be downloaded at
   http://www.netlib.org/clapack/
 contains a copy of f2c.  This copy copy is basically identical but it includes
 two files which are missing in the source which was the base for the libf2c2
 package.  Since the functions inside these files are used inside the test suite
 of clapack the files were added here to avoid code duplication.

--- /dev/null
+++ b/i_ceiling.c
@@ -0,0 +1,36 @@
+#include "f2c.h"
+
+#ifdef KR_headers
+integer i_sceiling(x) real *x;
+#else
+#ifdef __cplusplus
+extern "C" {
+#endif
+integer i_sceiling(real *x)
+#endif
+{
+#define CEIL(x) ((int)(x) + ((x) > 0 && (x) != (int)(x)))
+
+    return (integer) CEIL(*x);
+}
+#ifdef __cplusplus
+}
+#endif
+
+
+#ifdef KR_headers
+integer i_dceiling(x) doublereal *x;
+#else
+#ifdef __cplusplus
+extern "C" {
+#endif
+integer i_dceiling(doublereal *x)
+#endif
+{
+#define CEIL(x) ((int)(x) + ((x) > 0 && (x) != (int)(x)))
+
+    return (integer) CEIL(*x);
+}
+#ifdef __cplusplus
+}
+#endif
--- /dev/null
+++ b/i_len_trim.c
@@ -0,0 +1,22 @@
+#include "f2c.h"
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifdef KR_headers
+integer i_len_trim(s, n) char *s; ftnlen n;
+#else
+integer i_len_trim(char *s, ftnlen n)
+#endif
+{
+  int i;
+
+  for(i=n-1;i>=0;i--)
+    if(s[i] != ' ')
+      return i + 1;
+
+  return(0);
+}
+#ifdef __cplusplus
+}
+#endif
